spec/routing/file_routing_spec.rb

```diff
@@ -5,6 +5,49 @@
     expect(get: '/file/abc/def.pdf').to route_to('file#show', id: 'abc', file_name: 'def.pdf')
   end
 
+  context '#show (download): filename with' do
+    it 'chars not requiring URI escaping' do
+      filename = "(no_escape_needed):;=&$*.-_+!,~'.ext"
+      expect(get: "/file/oo000oo0000/#{filename}").to route_to(
+        'file#show', id: 'oo000oo0000', file_name: filename)
+    end
+
+    it 'some chars requiring URI escaping' do
+      filename = 'escape_needed {} @#^ %|\"`.ext'
+      expect(get: "/file/oo000oo0000/#{URI.escape(filename)}").to route_to(
+        'file#show', id: 'oo000oo0000', file_name: filename)
+    end
+
+    it 'square brackets must be url escaped' do
+      filename = 'foo[brackets].bar.pdf'
+      escaped_filename = 'foo%5Bbrackets%5D.bar.pdf' # URI.escape doesn't do square brackets
+      expect(get: "/file/oo000oo0000/#{escaped_filename}").to route_to(
+        'file#show', id: 'oo000oo0000', file_name: filename)
+    end
+
+    it 'question mark must be url escaped' do
+      filename = 'foo?.pdf'
+      escaped_filename = 'foo%3F.pdf' # URI.escape doesn't do question mark
+      expect(get: "/file/oo000oo0000/#{escaped_filename}").to route_to(
+        'file#show', id: 'oo000oo0000', file_name: filename)
+    end
+
+    it 'ü must be url escaped' do
+      skip('problem writing test:  ü decodes to \xC3\xBC')
+      filename = "fü.pdf"
+      escaped_filename = URI.escape(filename) # becomes %C3%BC
+      expect(get: "/file/oo000oo0000/#{escaped_filename}").to route_to(
+        'file#show', id: 'oo000oo0000', file_name: filename)
+    end
+
+    it '%20 in actual name (from ck155rf0207)' do
+      skip('problem writing test:  %20 becomes space, over-aggressive param decoding?')
```

Comment:
definitely very interested to see whether this happens IRL, because i find it super-surprising.  and as i noted in my recent brain dump comment, the thing that you think would be a crude console approximation of what's happening does what you'd expect (URL encode the string, URL decode the string, get the string back).